src/components/Products/SidebarFilterView/SidebarFilter.jsx

```diff
@@ -0,0 +1,119 @@
+import './sidebarFilter.css';
+import {useFilter} from "../../../contexts/filterContext";
+
+const SidebarFilter = () => {
+    const { state: {categories, priceRange, products}, dispatch }  = useFilter();
+    const stars = [1,2,3,4];
+
+    const changeHandler = (filterType, filterValue, e) => {
+        const type = filterType;
+        const payload = 
+        filterType === 'CATEOGRY' 
+            ? { [filterValue]: e.target.checked }
+            : filterValue;
+        dispatch({type : type, payload: payload}); 
+    }
+    return (
+        <div className="product-sidebar mr-6">
+            <aside className="side-navbar">
+                <header className="nav-header py-4">
+                    <div className="px-4 bold">Filters</div>
+                    <a href="" className="link-text-primary px-6"
+                    onClick = {() => changeHandler("CLEAR", products)}>clear</a>
+                </header>
+                <div className="side-navbar">
+                    <div className="price-filter">
+                        <div className="list-title px-4 my-2 bold">Price</div>
+                        <div className="slider px-4">
+                            <div className="range">
+                            <datalist id="tickmarks">
+                                <option value="300" label="300"></option>
+                                <option value="600" label="600"></option>
+                                <option value="900" label="900"></option>
+                                <option value="1200" label="1200"></option>
+                            </datalist>
+                            <input 
+                                className="rangeInput"
+                                type="range" 
+                                id="price" 
+                                name="price"
+                                min="300" 
+                                max="1299"
+                                value={priceRange}
+                                // step='300'
+                                onChange={(e) =>  changeHandler('PRICE', e.target.value)}
+                            />
+                            </div>
+                        </div>
+                    </div>
+                    <div className="rating-filter">
+                        <div className="list-title px-4 my-2 bold">Rating</div>
+                        <div className="checklist">
+                            {stars.reverse().map(star => {
```

Comment:
`reverse` is not a pure function since it does the reversal in place. Maybe you could try to avoid using this? or spread and then map?

Reply:
I'll will look for another way to do it. Thanks for reminding me that.